Remotion/Web/Core/UI/Controls/NavigationCommand.cs

```diff
@@ -29,22 +31,33 @@ namespace Remotion.Web.UI.Controls
   /// <summary> A <see cref="NavigationCommand"/> defines an action the user can invoke when navigating between pages. </summary>
   public class NavigationCommand : Command
   {
+    private readonly IFallbackNavigationUrlProvider _fallbackNavigationUrlProvider;
+
     public NavigationCommand ()
-        : this(CommandType.Href, GetWebSecurityAdapter(), GetWxeSecurityAdapter())
+        : this(CommandType.Href, GetWebSecurityAdapter(), GetWxeSecurityAdapter(), GetFallbackNavigationUrlProvider())
     {
     }
 
     public NavigationCommand (CommandType defaultType)
-        : this(defaultType, GetWebSecurityAdapter(), GetWxeSecurityAdapter())
+        : this(defaultType, GetWebSecurityAdapter(), GetWxeSecurityAdapter(), GetFallbackNavigationUrlProvider())
     {
     }
 
     public NavigationCommand (
         CommandType defaultType,
         [CanBeNull] IWebSecurityAdapter? webSecurityAdapter,
-        [CanBeNull] IWxeSecurityAdapter? wxeSecurityAdapter)
+        [CanBeNull] IWxeSecurityAdapter? wxeSecurityAdapter,
+        IFallbackNavigationUrlProvider fallbackNavigationUrlProvider)
         : base(defaultType, webSecurityAdapter, wxeSecurityAdapter)
     {
+      ArgumentUtility.CheckNotNull("fallbackNavigationUrlProvider", fallbackNavigationUrlProvider);
+
+      _fallbackNavigationUrlProvider = fallbackNavigationUrlProvider;
+    }
+
+    private static IFallbackNavigationUrlProvider GetFallbackNavigationUrlProvider ()
+    {
+      return SafeServiceLocator.Current.GetInstance<IFallbackNavigationUrlProvider>();
     }
```

Comment:
Static methods should be above the instance members. In this case, it should be the first member in this class.

Reply:
This one here's a false positive: the private static method is just a private method without state. These methods remain with the instance methods. 

Historically, we had these methods as regular instance methods but R# recommends making them static and it was decided that it's good practice to clearly state a method isn't relying on state. So, private methods that are only used from instance methods get placed with the other private instance methods and only private static methods that are used from public/protected static methods are collected at the top of the class.